### PR TITLE
Add subgraph for delegation history

### PR DESCRIPTION
**Pull request by adamgoth**
* Adds subgraph configuration
* Updates `fetchDelegatedTo` to use subgraph
* Update factory contarct for new factory
* Fix ENS fetch on wrong network and other type issues

### CODE REVIEW

**Review by torhector2 — APPROVED** [review]

File: modules/gql/gql.constants.ts [hunk 1]
```diff
@@ -12,6 +12,15 @@ export const STAGING_MAINNET_SPOCK_URL = 'https://pollingdb2-mainnet-staging.mak
...
 export const TENDERLY_SPOCK_URL = 'https://pollingdb2-tenderly-staging.makerdao.com/api/v1';
 
+/* Subgraph URLs */
+
+// const usePrivateSubgraph = process.env.USE_PRIVATE_SUBGRAPH === 'true';
+// const permission = usePrivateSubgraph ? 'private' : 'public';
+export const TENDERLY_SUBGRAPH_URL =
+  'https://query-subgraph-staging.sky.money/private/subgraphs/name/jetstreamgg/subgraph-testnet';
+export const MAINNET_SUBGRAPH_URL =
+  'https://query-subgraph-staging.sky.money/private/subgraphs/name/jetstreamgg/subgraph-testnet';
```
Comment (hernandoagf) [inline modules/gql/gql.constants.ts:22]: Shouldn't this be `/subgraph-mainnet`?

Comment (adamgoth) [reply modules/gql/gql.constants.ts]: Good catch!

**Review by tyler17 — APPROVED** [review]
Looks great. I noticed the vote histories & last voted dates look different than prod, but it's the same thing on staging, so it must be an issue with the staging mainnet spock being behind or stuck for votes.

**Review by tyler17 — COMMENTED** [review]
Actually, I'm noticing some issues in the history:
1. when I click to expand a delegation, it'll throw an error
2. it says N/A for all delegate contract expirations
3. When I compare the history for this address: 0xf65475e74c1ed6d004d5240b06e3088724dfda5d, I don't see the delegation events for BLUE or vigilant

Edit: here's a PR to fix these issues: https://github.com/makerdao/governance-portal-v2/pull/923/files

**Pull request merged**